src/pages/evm/swap.svelte

```diff
@@ -1,146 +1,122 @@
-
-    const ethAccount: Writable<EthAccount | null> = writable(null)
-    let currentTab = 'native'
-    let provider: ethers.providers.Web3Provider
-
-    async function switchNetwork() {
-        await window.ethereum
-            .request({
-                method: 'wallet_switchEthereumChain',
-                params: [{chainId: '0x4571'}],
-            })
-            .catch(async (e: {code: number}) => {
-                if (e.code === 4902) {
-                    await window.ethereum.request({
-                        method: 'wallet_addEthereumChain',
-                        params: [
-                            {
-                                chainId: '0x4571',
-                                chainName: 'EOS EVM Network',
-                                nativeCurrency: {name: 'EOS', symbol: 'EOS', decimals: 18},
-                                rpcUrls: ['https://api.evm.eosnetwork.com/'],
-                                blockExplorerUrls: ['https://explorer.evm.eosnetwork.com'],
-                            },
-                        ],
-                    })
-                }
-            })
-    }
+    import {transferNativeToEvm, transferEvmToNative, connectEthWallet} from '~/lib/evm'
 
-    async function connectWallet() {
-        if (window.ethereum) {
-            provider = new ethers.providers.Web3Provider(window.ethereum)
-            let networkId = await provider.getNetwork()
-            if (networkId.chainId !== 17777) {
-                await switchNetwork()
-                networkId = await provider.getNetwork()
-            }
+    import Page from '~/components/layout/page.svelte'
+    import Form from './swap/form.svelte'
+    import Confirm from './swap/confirm.svelte'
+    import Success from './swap/success.svelte'
+    import Error from './swap/error.svelte'
+    import type {Token} from '~/stores/tokens'
+
+    type AccountType = 'Native' | 'EVM'
+
+    let step = 'form'
+    let amount: string = ''
+    let errorMessage: string | undefined
+    let from: Token | undefined
+    let to: Token | undefined
+    let nativeTransactResult: TransactResult | undefined
+    let evmTransactResult: ethers.providers.TransactionResponse | undefined
+
+    async function transfer() {
+        if (!$evmAccount) {
+            return (errorMessage = 'An evm session is required.')
+        }
 
-            await window.ethereum.request({method: 'eth_requestAccounts'})
-            const signer = provider.getSigner()
-            ethAccount.set(EthAccount.from(await signer.getAddress()))
-        } else {
-            alert('You need to install Metamask')
+        try {
+            if (from?.name === 'EOS') {
+                nativeTransactResult = await transferNativeToEvm({
+                    nativeSession: $activeSession!,
+                    evmAccount: $evmAccount,
+                    amount,
+                })
+            } else {
+                evmTransactResult = await transferEvmToNative({
```

Comment:
Are you able to get this to work? I can't transfer from EVM -> Native, all I get is this:

<img width="1136" alt="image" src="https://github.com/greymass/unicove/assets/677686/ac343329-804a-4e30-bfb4-658b2f6bd0db">